### DISCUSSION TITLE
v4.1.3

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#136] [Announcements]
## What's Changed
* Accommodate the removal of API call GetUserUsageAllowance by @autoSteve
* Halve retry delays by @autoSteve
* Readme improvements by @autoSteve

### API change
Solcast recently removed a deprecated API endpoint, which was used to inquire about API usage limit and consumption for the day. This was causing errors to be logged on integration startup.

To work around this, and still keep track of API usage we have introduced a new configuration option, which is set when entering your API key.

Existing users upgrading should see little difference. In this case the API limit is loaded from the usage cache file and populates the new option. (If you're curious, check out the integration 'CONFIGURE' | 'Solcast API key' and you'll see the new option.) New users will be presented with a default of ten on install, and hopefully do not think this option is a "magic pudding" and decide to set it to 1,000...

### Swifter retries
v4.1.2 doubled the number of attempts to fetch forecasts, and it did this over a thirty minute period. This was really too long, so the number of tries in this release is the same, but the time frame halved to a maximum of fifteen minutes.

**Full Changelog**: https://github.com/BJReplay/ha-solcast-solar/compare/v4.1.2...v4.1.3

**Comment by autoSteve** [comment]
Notifying the usual early-adopting-and-reporting folk. @dtswk, @EddieSpaghettie, @ffm777, @gcoan, @hwb45, @iainfogg, @isorin, @jeffb1974, @kernow, @ladaan, @Markyt77, @Nilogax, @Noodleyman, @ProphetOfDoom, @rholligan, @Unlik and @wimdebruyn.

**Reply by gcoan** [reply]
I'm away for the weekend shortly so won't upgrade until I'm back @autoSteve but all good so far on 4.1.2

(Flying to the Isle of Wight, 2 miles? of open sea)

**Reply by autoSteve** [reply]
Friday releases are fraught, @gcoan. Great for me, because I can quickly respond to issues, but not great because folks go away.

Enjoy the Isle of Wight in Summer! The journey will be a bit blah, presumably in an ultra-light, but the dest is spekky.

**Reply by gcoan** [reply]
Oh yes, flying there, straight across the naval dockyards in Portsmouth harbour.  Much quicker (and cheaper) than driving and taking a ferry.   Apparently the most expensive ferry crossing in the world in terms of cost per mile (£190 for a return with car)

**Reply by isorin** [reply]
On holidays until the end of August, is it still working the old way? I don't want to upgrade while remote.

**Reply by autoSteve** [reply]
Continuous improvement, @isorin. Solcast removed a deprecated API call, which is the whole "how much have I used" thing. Should see changes to startup only as cache will deal, so I would not bother while you're absent.

**Reply by autoSteve** [reply]
~160 wonderful humans are on board so far without incident, and it's nearing my bed time, so that's welcome.

**Reply by Nilogax** [reply]
Just back from holiday so I've got no excuse to dodge the upgrade ;) No apparent issues, has updated forecast as expected.

**Reply by rholligan** [reply]
All good with 4.1.3 over here 🙏

**Reply by iainfogg** [reply]
Installed when you messaged, and no issues here

**Reply by gcoan** [reply]
View over the Solent heading to the Isle of Wight
![image](https://github.com/user-attachments/assets/8cba7c30-edba-48f5-9483-135cefe97421)

Needles lighthouse 
![image](https://github.com/user-attachments/assets/0eb71a18-0f66-4ab5-9b81-caae48048dd2)

**Comment by wimdebruyn** [comment]
4.1.3 in and no issues. API usage tracking working as usual.

Looking forward to the Springbok and Wallabies game later today!

**Comment by gcoan** [comment]
Installed 4.1.3 just now

decided that as solcast is working so well I would remove the debug setting from configuration.yaml:

```
#  logs:
#    custom_components.solcast_solar: debug
```

to cut down on the amount of logging, leaving the default warning level logging

was quite surprised after I restarted that there was absolutely no logging from the solcast integration at all, not even that it was starting or there was a schema upgrade that it did.

as far as I can tell its working OK, no errors to indicate otherwise.  Going into the integration configuration I can see the new 10 API limit has been populated

**Reply by autoSteve** [reply]
Nice confidence.

Yep. All startup/normal running logs at 'INFO' level. So if you're at warning it's gonna be, "management by exception".

**Reply by gcoan** [reply]
As I read it (its not well written), the default level for logger is warning https://www.home-assistant.io/integrations/logger/

I'll try changing it to info for all integrations and see how buried I get. Just wondered whether some of the startup messages for Solcast should be written as warning rather than info?

**Reply by autoSteve** [reply]
Yeah, I'm not big on the whole warn-for-the-sake-of-warning thing. So sorry. That will not be done under my watch.

I love a clean log, and unless it's this integration things get warning level. (And like you, even this integration may be demoted to warning for my prod rig.) Things should _just work_, and info be generally too chatty for many integrations in my experience (this integration tries to not be too chatty).

Then again, a warning about, "If you get any issues with this integration, blah, blah" could be fun.

**Reply by autoSteve** [reply]
> its not well written

That was _absolutely_ written by a software developer, @gcoan. 100% factual, yet 75% readable. 😂

**Comment by robinhood-code** [comment]
Just a very minor suggestion perhaps the text  'CONFIGURE' | 'Solcast API key'  can be changed to  'CONFIGURE' | 'Solcast API key **& Quota**'

**Reply by autoSteve** [reply]
This is a good suggestion, and is kind of done translations included. Readme and images to go.

<img width="378" alt="image" src="https://github.com/user-attachments/assets/1dab7141-34e6-4c0d-82c8-4158eaa95a8a">

**Reply by autoSteve** [reply]
PR #143 merged

**Reply by robinhood-code** [reply]
perfect, thanks !

**Comment by EddieSpaghettie** [comment]
Works well. Only one odity it showed my forecast for the next hour (this morning before the sun came out) was -0.5W but not an issue

**Reply by autoSteve** [reply]
A mathematical quirk of the splines that are now used. This interests me because there is code specifically to exclude negative values.

Thanks for noticing it.

**Reply by EddieSpaghettie** [reply]
No problem. ill post it tomorrow when it happens again. didnt take a screen shot

**Reply by autoSteve** [reply]
I got nothin'...

```
if math.copysign(1.0, self.fc_moment['all'][_data_field][i]) < 0:
    self.fc_moment['all'][_data_field][i] = 0.0 # Suppress negative values
```

**Comment by EddieSpaghettie** [comment]
Also not sure if i need to enable something but my graph from readme) doesnt show the forecast grey output as in the screenshot
![image](https://github.com/user-attachments/assets/48140449-0f3a-4001-825f-7c5d49a6e431)

**Reply by EddieSpaghettie** [reply]
Forecast and forecast 10

**Reply by autoSteve** [reply]
~~Do you have the detailed forecast attribute enabled in `CONFIGURE`?~~

**Reply by autoSteve** [reply]
I will bet it is a numbers scale thing. Your generation is in Watts (even though legend says kW). The forecast will be kW, so the forecast will be there, just really, really small.

**Reply by EddieSpaghettie** [reply]
I just ticket the last two boxes but still not showing. Ill try the watss/kw as per the doc and see if that helps
![image](https://github.com/user-attachments/assets/daed3b22-451e-46cc-b9d3-1750e2bef7c9)

**Reply by EddieSpaghettie** [reply]
Im not sure where to add this : transform: "return x / 1000;"

**Reply by autoSteve** [reply]
<img width="388" alt="image" src="https://github.com/user-attachments/assets/e3597449-c0eb-4b6a-bba9-4e279c2784fc">

**Reply by EddieSpaghettie** [reply]
![image](https://github.com/user-attachments/assets/caaf6b77-49db-4549-8642-908945f4aca4)
Still not

**Reply by EddieSpaghettie** [reply]
And the header broke where it shows today forecast 10%

**Reply by EddieSpaghettie** [reply]
![image](https://github.com/user-attachments/assets/1e5e9393-f5b7-4675-af94-c44a503efffe)

**Reply by autoSteve** [reply]
Add it to your _solar generation_ sensor, not the forecast...

**Reply by EddieSpaghettie** [reply]
That worked:
![image](https://github.com/user-attachments/assets/51a56295-5569-4956-89af-387b019fbb49)
Thanks

**Reply by BJReplay** [reply]
All these people with 25kWh and 50kWh forecasts, and I'm sitting here with my 15kWh forecast and 12kWh actuals.

Still, energy used likely to top out around 13kWh, and the new battery charge algorithm is working...

![image](https://github.com/user-attachments/assets/5361e33e-7d1a-4a4e-9254-35b69a47f5ac)

At this rate, I'll have paid off my battery in 300 years :)

**Reply by EddieSpaghettie** [reply]
Lol. Sorry man. Drop your efficiency number on solcast website to get your predicted and actual closer together. Mine is on 93% and is pretty close. Dont think it will make the 300 years less but...

**Reply by autoSteve** [reply]
Lol @BJReplay. Much roof here, so I just said, "Give me as many panels as I'm allowed." 56kWh forecast today, which is not bad for Winter. ;)

**Reply by BJReplay** [reply]
> Drop your efficiency number on solcast website to get your predicted and actual closer together.

Mine's usually pretty good, TBH, cloud just hung around longer in the morning. But I should put the drone up and check if they need cleaning (top of a flat roof on the second storey).

**Reply by gcoan** [reply]
> All these people with 25kWh and 50kWh forecasts, and I'm sitting here with my 15kWh forecast and 12kWh actuals.

I'm usually in that first group, forecasting 60/70kWh and achieving 50kWh or so.  Even a cloudy summer's day like the day after tomorrow is forecast to be 30kWh.

But today the forecast is just 23kWh with the 10% PV at 9.4kWh.  Peak generation 1280W!

But it's very windy as well so the grid import price was negative for most of the night and apart from the morning breakfast 'bulge' is negative for most of the day as well.  Not massively negative, just 1 or 2 pence per kWh, but my export remains fixed at 15p so I will make money by running the tumble dryer and filling and discharging the battery.
Crazy world.

Might try dropping my Solcast efficiency to see if it improves the accuracy of my forecast.  I still consistently under-achieve every day

**Reply by BJReplay** [reply]
> Give me as many panels as I'm allowed.

If I was to go again, replacing a 13 year old system, I could probably fit 13.2kW on my roof, vs 4.85kW as installed back in 2011

That would turn today into at 40kWh day.

**Reply by gcoan** [reply]
> But I should put the drone up and check if they need cleaning (top of a flat roof on the second storey).

Depends on your weather of course, but in rainy Europe with panels on 30 degree or so pitched roofs, I'm pretty sure that cleaning the panels is not needed.  Ever.

I had my first set of 16 panels (4kW array) fitted 7 years ago and they never once got washed or cleaned. I recorded quarterly and annual generation figures and there was no discernible reduction over the years due to dirt or aging.  Some years better, some years slightly worse, but the later years overall were better than the early years.

When I had the scaffolding up for installation of the further 28 panels (10.4kW) I took the opportunity to inspect and wash the panels.  There were a few spots of lichen on a few places on the panels but not appreciably so.

So I'm of the view that washing the solar panels is one of these scam sales that door to door salesmen like to make.

If your panels are installed almost horizontal and/or lots of dusty weather then maybe, but not for me

**Reply by autoSteve** [reply]
> If I was to go again

Do it. The current panels won't owe you anything by now because you were probably on a premium feed-in tariff. And more generation will help fill the battery on more days of the year helping it pay for itself, assuming big enough battery (I had twelve straight days earlier this month in Winter where I consumed zero grid kWh, and this place doesn't go under 0.8kW draw every hour even if I'm not here. Month to date I pulled a small amount on just 3 out of 22 days, all off-peak. 27.8kWh bat capacity).

**Reply by BJReplay** [reply]
> So I'm of the view that washing the solar panels is one of these scam sales that door to door salesmen like to make.

In general, yes 

> If your panels are installed almost horizontal and/or lots of dusty weather then maybe, but not for me. 

They are - effectively flat. 

With 13 years of monitoring, I am pretty happy with the long term performance, but spotting short term degradation is harder. 

When we have had dusty weather (we have), followed by a little bit of rain (we have) I end up with mud on the lower edges of the panels and that is enough to make a difference. 

Normally dusty weather on its own, or followed by decent rain is fine.

**Reply by Noodleyman** [reply]
> er then maybe, but not for me

I disagree with the panel washing. I'm lucky that I live in a bungalow and am quite agile and happy to fix most things myself. I wash my panels anually. the level of dust and bidr crap they pick up through the summer is crazy. It does make a difference, 5-10% depending on how bad they are.  There's nothing quite like hauling a pressure washer onto the roof and the neighbours wondering wtf you are doing.

**Reply by robinhood-code** [reply]
I think you shouldn’t be using pressure washer to wash the panels ?

**Reply by Noodleyman** [reply]
works fine, power adjustable so it's on a soft/low power wash mode. they are essentially a glass sheet over the panel so it does a good job of getting all the greened up lichen off.  I wouldn't use a full power washer on them.

**Reply by robinhood-code** [reply]
yah, I understand the glass bit but the worry is more about weakening the seal and causes water ingress

**Reply by Noodleyman** [reply]
> yah, I understand the glass bit but the worry is more about weakening the seal and causes water ingress

not sure if all are like ours, but the outer frame covers the edges as an overlap, so you can't actually get to the seals with the washer as they are under the frame

**Reply by BJReplay** [reply]
I use a soft brush - a truck washing brush on an extending aluminum pole with a hose fitting - which is enough to dislodge the caked-on mud and flush it away.

**Comment by EddieSpaghettie** [comment]
![Screenshot_20240822_060458_Home Assistant](https://github.com/user-attachments/assets/5d751df6-ab2f-4d2a-88a0-8afa45a10af1)

Here is the negative number I was refereing to

**Reply by autoSteve** [reply]
Thanks for that, @EddieSpaghettie. I worked out why it is and I've added a commit to address it. Well spotted.

Once that is released, and on days where this occurs, you will notice the custom hour sensor rise above zero for a little while, fall back to zero, then go normal for the day. This is being caused by a minor spline 'bounce' that I am yet to address. If I address it prior to the next release then you won't notice it at all.

**Reply by EddieSpaghettie** [reply]
Really doesnt bother me as I dont look at that often. Just thought Ill mention it. Thanks for fixing it and your hard work on this project!

**Reply by autoSteve** [reply]
Pleasure, @EddieSpaghettie.

The minor spline bounce is committed, plus in looking I identified another minor issue waiting to be found and that is fixed, so really great call out.
